### PR TITLE
#161255356 - Social login 

### PR DESCRIPTION
**Pull request by Jwesh**
#### What does this PR do?
* Enable Social login via Google, Facebook and Twitter
#### Description of Task to be completed?
* Currently a new user cannot login with their pre-existing social media accounts. He/She can only register and or login with email password combinations
#### How should this be manually tested?
1. clone the repo and switch to ft-social-login-161255356
2. set environment and also configure database 
3. get token from google [developer playground](https://developers.google.com/oauthplayground/)
4. get token from facebook [developer console](https://developers.facebook.com/apps/541475206299733/roles/test-users/)

#### Any background context you want to provide?
* social login improves user experience while using the application.
#### What are the relevant pivotal tracker stories?
* Enable Social login via Google, Facebook and Twitter-[161255356](https://www.pivotaltracker.com/story/show/161255356)
#### Screenshots
screen shot 2018-11-01 at 14 13 19

### CODE REVIEW

**Review by AnguleMathias — APPROVED** [review]
Looks great! 
Check on test coverage.

**Review by meshnesh — CHANGES_REQUESTED** [review]
Write test for the added code to increase the coverage

**Review by njeri-ngigi — APPROVED** [review]

**Review by maxwellgithinji — APPROVED** [review]
LGTM

**Review by AnguleMathias — APPROVED** [review]

**Review by meshnesh — APPROVED** [review]

**Review by meshnesh — CHANGES_REQUESTED** [review]
Fix the failing test.

File: authors/apps/authentication/backends.py [hunk 1]
```diff
@@ -12,58 +12,71 @@
...
         """
         return 'Bearer'
+
+    def encode_token(self, email):
+        """
+        This method gerate token by encoding registered user
+        email andress
+        """
+        payload = {
+            'email': email,
```
Comment (meshnesh) [inline authors/apps/authentication/backends.py]: Is the token supposed to be encoded with the `email` or `user_id`?

**Review by meshnesh — APPROVED** [review]

**Pull request merged**